auth/src/main/java/org/aerogear/mobile/auth/credentials/JwksManager.java

```diff
@@ -110,28 +111,35 @@ public void fetchJwks(@NonNull final KeycloakConfiguration keycloakConfiguration
             JsonWebKeySet jwks = null;
             JwksException error = null;
             // this is invoked on a background thread.
-            if (response.getStatus() == 200) {
-                String jwksContent = response.stringBody();
-                try {
-                    jwks = new JsonWebKeySet(jwksContent);
-                } catch (JoseException e) {
-                    jwks = null;
-                    error = new JwksException(e);
-                    logger.warning("failed to parse JWKS key content. content = " + jwksContent);
+            try {
+                if (response.getStatus() == 200) {
+                    String jwksContent = response.stringBody();
+                    try {
+                        jwks = new JsonWebKeySet(jwksContent);
+                    } catch (JoseException e) {
+                        jwks = null;
+                        error = new JwksException(e);
+                        logger.warning("failed to parse JWKS key content. content = "
+                                        + jwksContent);
+                    }
+                    if (jwks != null) {
+                        persistJwksContent(keycloakConfiguration.getRealmName(), jwksContent);
+                    }
+                } else {
+                    logger.warning("failed to fetch JWKS from server. url = " + jwksUrl
+                                    + " statusCode = " + response.getStatus());
+                    error = new JwksException(jwksException);
```

Comment:
Why not simply throwing this exception and doing the `callback.onError` in the catch block?